### PR TITLE
Fold Strategy Updates

### PR DESCRIPTION
**Pull request by ngupta23**
#### Describe the changes you've made

Previously, time series experiment only accepted a string as input for fold_strategy. With this change, it can also accept a sktime compatible cv object directly. If an object is passed, then the `fold` and `fh` parameters in setup are ignored and are extracted from the cv object directly.

## Type of change

Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Updated unit tests

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

None

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/internal/pycaret_experiment/tabular_experiment.py [hunk 1]
```diff
@@ -906,17 +911,25 @@ def setup(
...
+                fold_strategy in possible_fold_strategy
+                or is_sklearn_cv_generator(fold_strategy)
+            ):
+                raise TypeError(
+                    f"fold_strategy parameter must be either a scikit-learn compatible CV generator object or one of {', '.join(possible_fold_strategy)}."
+                )
+        elif self._ml_usecase == MLUsecase.TIME_SERIES:
+            if not (
+                fold_strategy in possible_time_series_fold_strategies
+                or is_sklearn_cv_generator(fold_strategy)
```
Comment (TremaMiguel) [inline pycaret/internal/pycaret_experiment/tabular_experiment.py:928]: `is_sklearn_cv_generator` from which module is imported ?

File: pycaret/tests/test_time_series_splitters.py [hunk 2]
```diff
@@ -50,6 +50,43 @@ def test_setup_initialization(fold, fh, fold_strategy, load_data):
         assert exp_name.fold_generator.step_length == fh  # Since fh is an int
 
 
+def test_setup_pass_cv_object(load_data):
+
```
Comment (TremaMiguel) [inline pycaret/tests/test_time_series_splitters.py:54]: I think we could add docstrings to test to get a general idea of what we want to test.

Comment (TremaMiguel) [inline pycaret/tests/test_time_series_splitters.py:54]: I think we could add docstrings to tests to get a general idea of what we want to test.

Comment (TremaMiguel) [inline pycaret/tests/test_time_series_splitters.py:54]: I think we could add docstrings to tests to get a general idea of what we want to test.

**Review by TremaMiguel — APPROVED** [review]

File: pycaret/internal/pycaret_experiment/tabular_experiment.py (hunk 1 above)
Comment (ngupta23) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py:928]: It is defined in `pycaret` itself. Just looks to see if there is a split method in the object.

File: pycaret/tests/test_time_series_splitters.py (hunk 2 above)
Comment (ngupta23) [reply pycaret/tests/test_time_series_splitters.py:55]: Good idea. I have added it now.

**Pull request merged**